content/actions/automating-builds-and-tests/building-and-testing-java-with-gradle.md

```diff
@@ -65,21 +65,26 @@ We recommend that you have a basic understanding of Java and the Gradle framewor
    jobs:
      build:
        runs-on: ubuntu-latest
-
        steps:
        - uses: {% data reusables.actions.action-checkout %}
-       - name: Set up JDK 11
+       - name: Set up JDK 17
          uses: {% data reusables.actions.action-setup-java %}
          with:
-           java-version: '11'
+           java-version: '17'
            distribution: 'temurin'
+
+       - name: Setup Gradle
+         uses: gradle/actions/setup-gradle@417ae3ccd767c252f5661f1ace9f835f9654f2b5 # v3.1.0
+
        - name: Build with Gradle
-         uses: gradle/gradle-build-action@bd5760595778326ba7f1441bcf7e88b49de61a25 # v2.6.0
-         with:
-           arguments: build
+         run: ./gradlew build
    ```
 
 {%- endif %}
+{% data reusables.actions.gradle-workflow-steps %}
+1. The "Build with Gradle" step executes the `build` task using the [Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html).
```

Comment:
Just a passing note since you've made a lot of contributions to these docs @bigdaz : If you are interested, you are welcome to open a separate PR to move the explanatory steps to be inline with the code.

For more info about how to do this and an example of how code annotations render, see https://docs.github.com/en/contributing/writing-for-github-docs/annotating-code-examples

Reply:
Wow, that's nice. I'll take a look at incorporating annotations into the code.